src/coreclr/src/jit/optcse.cpp

```diff
@@ -2678,17 +2773,37 @@ class CSE_Heuristic
             cseSsaNum               = m_pCompiler->lvaTable[cseLclVarNum].lvPerSsaData.AllocSsaNum(allocator);
         }
 
-#ifdef DEBUG
         // Verify that all of the ValueNumbers in this list are correct as
         // Morph will change them when it performs a mutating operation.
         //
         ValueNum firstVN = ValueNumStore::NoVN;
         ValueNum currVN;
-        bool     allSame = true;
+        bool     setRefCnt  = true;
+        bool     allSame    = true;
+        bool     isConstCSE = ((dsc->csdHashKey & TARGET_SIGN_BIT) != 0);
+        unsigned maxItem    = 0;
+
+        struct constCSE_entry
```

Comment:
Please add a `const unsigned itemsMaxSize = 8` with a comment why `8` is chosen.